images/build-e2e/README.md

```diff
@@ -1,50 +1,87 @@
 # Overview
 
-The container includes the e2e binary for all 3 platforms plus the required resources to run it.  
+This image contains the e2e suite of tests, and it is intended to run them agaisnt a target host, the logic to run the tests remotely is inherit from its base image: [deliverest](https://github.com/adrianriobo/deliverest), each platform and os has its own image.
 
-The container connects through ssh to the target host and copy the right binary for the platform, run e2e tests and pick the results and logs back.
+## Run
 
-## Envs
+The image version contains the specs about the plaftorm and the arch; then the test customization is made by the command executed within the image; as we can see the cmd should be defined depending on the platform:
 
-**PLATFORM**:*define target platform (windows, macos, linux).*
-**ARCH**:*define target arch (amd64, arm64). Default amd64
-**TARGET_HOST**:*dns or ip for the target host.*  
-**TARGET_HOST_USERNAME**:*username for target host.*  
-**TARGET_HOST_KEY_PATH**:*private key for user. (Mandatory if not TARGET_HOST_PASSWORD).*  
-**TARGET_HOST_PASSWORD**:*password for user. (Mandatory if not TARGET_HOST_KEY_PATH).*  
-**PULL_SECRET_FILE_PATH**: *pull secret file path (local to container).*  
-**BUNDLE_VERSION**:*(Mandatory if not BUNDLE_LOCATION). Testing agaisnt crc released version bundle version for crc released version.*
-**BUNDLE_LOCATION**:*(Mandatory if not BUNDLE_VERSION). When testing crc with custom bundle set the bundle location on target server.*  
-**RESULTS_PATH**:*(Optional). Path inside container to pick results and logs from e2e execution.*  
-**RESULTS_FILE**:*(Optional). File name for results xunit results. Default value: e2e.*  
-**CLEANUP_HOME**:*(Optional). Cleanup crc home folder or keep as it is to run test.*  
-**E2E_TAG_EXPRESSION**:*(Optional). Define e2e tag expression to select tests. If empty all tests available for the platform will be executed.*  
+* crc-e2e/run.ps1 ... (windows)
+* crc-e2e/run.sh ... (linux/darwin)
 
-## Samples
+And the execution is customized by the params addded, available params:
+
+* bundleLocation When testing a custom bundle we should pass the paht on the target host
+* e2eTagExpression To set an specific set of tests based on annotations 
+* targetFolder Name of the folder on the target host under $HOME where all the content will be copied
+* junitFilename Name for the junit file with the tests results
+* crcMemory Customize memory for the cluster to run the tests
+
+### Windows amd64
+
+```bash
+podman run --rm -d --name crc-e2e-windows \
+    -e TARGET_HOST=XXXX \
+    -e TARGET_HOST_USERNAME=XXXX \
+    -e TARGET_HOST_KEY_PATH=/data/id_rsa \
+    -e TARGET_FOLDER=crc-e2e \
```

Comment:
Does this `TARGET_FOLDER` override the `ASSETS_FOLDER` of the image? because as part of container file following is what we mention
```
COPY images/build-e2e/lib/${OS}/* ${ASSETS_FOLDER}/
```
and during the build of this container image from makefile I don't see passing a ASSETS_FOLDER so not sure how `run.ps1/run.sh` lands to this `TARGET_FOLDER`.

Reply:
no, TARGET_FOLDER takes effect on run time, and it makes reference to the folder path where we want to copy the required resources on the target host

The ASSETS_FOLDER is defined on build time, and it is being used by the base image `quay.io/rhqp/deliverest:v0.0.5`

Deliverest image has some common bash functions to ssh / scp and implement the logic:

* Copy everything on the container `ASSETS_FOLDER` to the remote target `TARGET_FOLDER`
* Run on the remote target the command of the container
* Copy back the folder from the target host at `TARGET_RESULTS` to the container path defined by `OUTPUT_FOLDER`
* Remove the `TARGET_FOLDER` on the target host

This logic is being used on almost every test execution done against a remote target, this is why it has been abstracted into Deliverest and now each image inheriting from it just need to define its specific workload per platform / arch

Reply:
Okay so this is part of https://github.com/adrianriobo/deliverest/blob/main/entrypoint.sh#L37 one.

Reply:
yeah exactly